<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models â€¢ ppcseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models">
<meta property="og:description" content="ppcseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ppcseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.8.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/introduction.html">Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stemangiola/ppcseq/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Probabilistic Outlier Identification for RNA
Sequencing Generalized Linear Models</h1>
                        <h4 data-toc-skip class="author">Stefano
Mangiola</h4>
            
            <h4 data-toc-skip class="date">2023-09-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stemangiola/ppcseq/blob/HEAD/vignettes/introduction.Rmd" class="external-link"><code>vignettes/introduction.Rmd</code></a></small>
      <div class="hidden name"><code>introduction.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Relative transcript abundance has proven to be a valuable tool for
understanding the function of genes in biological systems. For the
differential analysis of transcript abundance using RNA sequencing data,
the negative binomial model is by far the most frequently adopted.
However, common methods that are based on a negative binomial model are
not robust to extreme outliers, which we found to be abundant in public
datasets. So far, no rigorous and probabilistic methods for detection of
outliers have been developed for RNA sequencing data, leaving the
identification mostly to visual inspection. Recent advances in Bayesian
computation allow large-scale comparison of observed data against its
theoretical distribution given in a statistical model. Here we propose
ppcseq, a key quality-control tool for identifying transcripts that
include outlier data points in differential expression analysis, which
do not follow a negative binomial distribution. Applying ppcseq to
analyse several publicly available datasets using popular tools, we show
that from 3 to 10 percent of differentially abundant transcripts across
algorithms and datasets had statistics inflated by the presence of
outliers.</p>
</div>
<div class="section level2">
<h2 id="installation-and-use">Installation and use<a class="anchor" aria-label="anchor" href="#installation-and-use"></a>
</h2>
<p>The input data set is a tidy representation of a differential gene
transcript abundance analysis</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stemangiola/ppcseq" class="external-link">ppcseq</a></span><span class="op">)</span></span></code></pre></div>
<p>To install:</p>
<p>Before install, for linux systems, in order to exploit
multi-threading, from R write:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fileConn</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">file</a></span><span class="op">(</span><span class="st">"~/.R/Makevars"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/writeLines.html" class="external-link">writeLines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"CXX14FLAGS += -O3"</span>,<span class="st">"CXX14FLAGS += -DSTAN_THREADS"</span>, <span class="st">"CXX14FLAGS += -pthread"</span><span class="op">)</span>, <span class="va">fileConn</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">close</a></span><span class="op">(</span><span class="va">fileConn</span><span class="op">)</span></span></code></pre></div>
<p>Multi-threading allows the sampling or variational bayes to share the
computation on multiple cores.</p>
<p>Then, install with</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ppcseq"</span><span class="op">)</span></span></code></pre></div>
<p>You can get the test dataset with</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="va">counts</span> </span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 394,821 Ã— 9</span></span></span>
<span><span class="co">##    sample  symbol   logCPM    LR   PValue        FDR value       W Label      </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> 10922PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   160 -<span style="color: #BB0000;">0.129</span>  High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> 10935PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   150 -<span style="color: #BB0000;">0.127</span>  High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> 10973PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   146 -<span style="color: #BB0000;">0.426</span>  High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 10976PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   347 -<span style="color: #BB0000;">0.016</span><span style="color: #BB0000; text-decoration: underline;">4</span> High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> 10985PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   175 -<span style="color: #BB0000;">0.135</span>  High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 11026PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   244  0.125  High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> 11045PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   399 -<span style="color: #BB0000;">0.089</span><span style="color: #BB0000; text-decoration: underline;">2</span> High       </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 11082PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74   100  0.261  Neoadjuvant</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> 11086PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74    37 -<span style="color: #BB0000;">0.132</span>  Neoadjuvant</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> 11103PP SLC16A12   1.39  41.1 1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-10</span> 0.000<span style="text-decoration: underline;">002</span>74    73  0.146  Neoadjuvant</span></span>
<span><span class="co">## <span style="color: #949494;"># â„¹ 394,811 more rows</span></span></span></code></pre>
<p>You can identify anrtefactual calls from your differential transcribt
anundance analysis, due to outliers.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import libraries</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.info.html" class="external-link">Sys.info</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">'sysname'</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Linux"</span><span class="op">)</span></span>
<span><span class="va">counts.ppc</span> <span class="op">=</span> </span>
<span>    <span class="va">counts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>is_significant <span class="op">=</span> <span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.0001</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/identify_outliers.html">identify_outliers</a></span><span class="op">(</span></span>
<span>        formula <span class="op">=</span> <span class="op">~</span> <span class="va">Label</span>,</span>
<span>        .sample <span class="op">=</span> <span class="va">sample</span>, </span>
<span>        .transcript <span class="op">=</span> <span class="va">symbol</span>,</span>
<span>        .abundance <span class="op">=</span> <span class="va">value</span>,</span>
<span>        .significance <span class="op">=</span> <span class="va">PValue</span>,</span>
<span>        .do_check <span class="op">=</span> <span class="va">is_significant</span>,</span>
<span>        percent_false_positive_genes <span class="op">=</span> <span class="fl">5</span>, </span>
<span>        approximate_posterior_inference <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        cores <span class="op">=</span> <span class="fl">1</span>, </span>
<span>        </span>
<span>        <span class="co"># This is ONLY for speeding up the Vignette execution</span></span>
<span>        draws_after_tail <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>The new posterior predictive check has been added to the original
data frame</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.info.html" class="external-link">Sys.info</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">'sysname'</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Linux"</span><span class="op">)</span></span>
<span><span class="va">counts.ppc</span> </span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 1 Ã— 4</span></span></span>
<span><span class="co">##   symbol   sample_wise_data   ppc_samples_failed tot_deleterious_outliers</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>                          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> SLC16A12 <span style="color: #949494;">&lt;tibble [21 Ã— 13]&gt;</span>                  0                        0</span></span></code></pre>
<p>The new data frame contains plots for each gene</p>
<p>We can visualise the top five differentially transcribed genes</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.info.html" class="external-link">Sys.info</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">'sysname'</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Linux"</span><span class="op">)</span></span>
<span><span class="va">counts.ppc_plots</span> <span class="op">=</span> </span>
<span>    <span class="va">counts.ppc</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/plot_credible_intervals.html">plot_credible_intervals</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.info.html" class="external-link">Sys.info</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">'sysname'</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Linux"</span><span class="op">)</span></span>
<span><span class="va">counts.ppc_plots</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">plot</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [[1]]</span></span></code></pre>
<p><img src="introduction_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ppcseq_1.8.1         rstan_2.21.8         ggplot2_3.4.2       </span></span>
<span><span class="co">## [4] StanHeaders_2.21.0-7 dplyr_1.1.1          knitr_1.42          </span></span>
<span><span class="co">## [7] BiocStyle_2.26.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] httr_1.4.5            tidyr_1.3.0           sass_0.4.5           </span></span>
<span><span class="co">##  [4] edgeR_3.40.2          foreach_1.5.2         jsonlite_1.8.4       </span></span>
<span><span class="co">##  [7] bslib_0.4.2           RcppParallel_5.1.7    posterior_1.4.1      </span></span>
<span><span class="co">## [10] distributional_0.3.2  highr_0.10            BiocManager_1.30.20  </span></span>
<span><span class="co">## [13] stats4_4.2.3          ggdist_3.2.1          tensorA_0.36.2       </span></span>
<span><span class="co">## [16] yaml_2.3.7            pillar_1.9.0          backports_1.4.1      </span></span>
<span><span class="co">## [19] lattice_0.21-8        glue_1.6.2            limma_3.54.2         </span></span>
<span><span class="co">## [22] arrayhelpers_1.1-0    digest_0.6.31         checkmate_2.1.0      </span></span>
<span><span class="co">## [25] colorspace_2.1-0      Matrix_1.5-4          htmltools_0.5.5      </span></span>
<span><span class="co">## [28] pkgconfig_2.0.3       svUnit_1.0.6          bookdown_0.33        </span></span>
<span><span class="co">## [31] purrr_1.0.1           scales_1.2.1          processx_3.8.1       </span></span>
<span><span class="co">## [34] tibble_3.2.1          generics_0.1.3        farver_2.1.1         </span></span>
<span><span class="co">## [37] cachem_1.0.7          withr_2.5.0           cli_3.6.1            </span></span>
<span><span class="co">## [40] magrittr_2.0.3        crayon_1.5.2          memoise_2.0.1        </span></span>
<span><span class="co">## [43] evaluate_0.20         ps_1.7.5              fs_1.6.1             </span></span>
<span><span class="co">## [46] fansi_1.0.4           doParallel_1.0.17     benchmarkme_1.0.8    </span></span>
<span><span class="co">## [49] pkgbuild_1.4.0        textshaping_0.3.6     tools_4.2.3          </span></span>
<span><span class="co">## [52] loo_2.6.0             prettyunits_1.1.1     lifecycle_1.0.3      </span></span>
<span><span class="co">## [55] matrixStats_0.63.0    stringr_1.5.0         munsell_0.5.0        </span></span>
<span><span class="co">## [58] locfit_1.5-9.7        callr_3.7.3           compiler_4.2.3       </span></span>
<span><span class="co">## [61] pkgdown_2.0.7         jquerylib_0.1.4       tidybayes_3.0.4      </span></span>
<span><span class="co">## [64] systemfonts_1.0.4     rlang_1.1.0           grid_4.2.3           </span></span>
<span><span class="co">## [67] iterators_1.0.14      labeling_0.4.2        rmarkdown_2.21       </span></span>
<span><span class="co">## [70] gtable_0.3.3          codetools_0.2-19      inline_0.3.19        </span></span>
<span><span class="co">## [73] abind_1.4-5           benchmarkmeData_1.0.4 R6_2.5.1             </span></span>
<span><span class="co">## [76] gridExtra_2.3         rstantools_2.3.1      fastmap_1.1.1        </span></span>
<span><span class="co">## [79] utf8_1.2.3            rprojroot_2.0.3       ragg_1.2.5           </span></span>
<span><span class="co">## [82] desc_1.4.2            stringi_1.7.12        parallel_4.2.3       </span></span>
<span><span class="co">## [85] Rcpp_1.0.10           vctrs_0.6.2           tidyselect_1.2.0     </span></span>
<span><span class="co">## [88] xfun_0.38             coda_0.19-4</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Stefano Mangiola.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
