<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models â€¢ ppcseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models">
<meta property="og:description" content="ppcseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ppcseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/introduction.html">Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stemangiola/ppcseq/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="introduction_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Probabilistic Outlier Identification for RNA Sequencing Generalized Linear Models</h1>
                        <h4 data-toc-skip class="author">Stefano Mangiola</h4>
            
            <h4 data-toc-skip class="date">2021-08-31</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stemangiola/ppcseq/blob/master/vignettes/introduction.Rmd" class="external-link"><code>vignettes/introduction.Rmd</code></a></small>
      <div class="hidden name"><code>introduction.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor" aria-hidden="true"></a>Introduction</h1>
<p>Relative transcript abundance has proven to be a valuable tool for understanding the function of genes in biological systems. For the differential analysis of transcript abundance using RNA sequencing data, the negative binomial model is by far the most frequently adopted. However, common methods that are based on a negative binomial model are not robust to extreme outliers, which we found to be abundant in public datasets. So far, no rigorous and probabilistic methods for detection of outliers have been developed for RNA sequencing data, leaving the identification mostly to visual inspection. Recent advances in Bayesian computation allow large-scale comparison of observed data against its theoretical distribution given in a statistical model. Here we propose ppcseq, a key quality-control tool for identifying transcripts that include outlier data points in differential expression analysis, which do not follow a negative binomial distribution. Applying ppcseq to analyse several publicly available datasets using popular tools, we show that from 3 to 10 percent of differentially abundant transcripts across algorithms and datasets had statistics inflated by the presence of outliers.</p>
</div>
<div id="installation-and-use" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-and-use" class="anchor" aria-hidden="true"></a>Installation and use</h1>
<p>The input data set is a tidy representation of a differential gene transcript abundance analysis</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ppcseq</span><span class="op">)</span></code></pre></div>
<p>To install:</p>
<p>Before install, for linux systems, in order to exploit multi-threading, from R write:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fileConn</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">file</a></span><span class="op">(</span><span class="st">"~/.R/Makevars"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/writeLines.html" class="external-link">writeLines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"CXX14FLAGS += -O3"</span>,<span class="st">"CXX14FLAGS += -DSTAN_THREADS"</span>, <span class="st">"CXX14FLAGS += -pthread"</span><span class="op">)</span>, <span class="va">fileConn</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">close</a></span><span class="op">(</span><span class="va">fileConn</span><span class="op">)</span></code></pre></div>
<p>Multi-threading allows the sampling or variational bayes to share the computation on multiple cores.</p>
<p>Then, install with</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ppcseq"</span><span class="op">)</span></code></pre></div>
<p>You can get the test dataset with</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"counts"</span><span class="op">)</span>
<span class="va">counts</span> </code></pre></div>
<pre><code>## # A tibble: 394,821 Ã— 9
##    sample  symbol   logCPM    LR   PValue        FDR value       W Label      
##    &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt; &lt;int&gt;   &lt;dbl&gt; &lt;chr&gt;      
##  1 10922PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   160 -0.129  High       
##  2 10935PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   150 -0.127  High       
##  3 10973PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   146 -0.426  High       
##  4 10976PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   347 -0.0164 High       
##  5 10985PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   175 -0.135  High       
##  6 11026PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   244  0.125  High       
##  7 11045PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   399 -0.0892 High       
##  8 11082PP SLC16A12   1.39  41.1 1.46e-10 0.00000274   100  0.261  Neoadjuvant
##  9 11086PP SLC16A12   1.39  41.1 1.46e-10 0.00000274    37 -0.132  Neoadjuvant
## 10 11103PP SLC16A12   1.39  41.1 1.46e-10 0.00000274    73  0.146  Neoadjuvant
## # â€¦ with 394,811 more rows</code></pre>
<p>You can identify anrtefactual calls from your differential transcribt anundance analysis, due to outliers.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Import libraries</span>


<span class="va">counts.ppc</span> <span class="op">=</span> 
    <span class="va">counts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>is_significant <span class="op">=</span> <span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="../reference/identify_outliers.html">identify_outliers</a></span><span class="op">(</span>
        formula <span class="op">=</span> <span class="op">~</span> <span class="va">Label</span>,
        .sample <span class="op">=</span> <span class="va">sample</span>, 
        .transcript <span class="op">=</span> <span class="va">symbol</span>,
        .abundance <span class="op">=</span> <span class="va">value</span>,
        .significance <span class="op">=</span> <span class="va">PValue</span>,
        .do_check <span class="op">=</span> <span class="va">is_significant</span>,
        percent_false_positive_genes <span class="op">=</span> <span class="fl">5</span>
    <span class="op">)</span></code></pre></div>
<p>The new posterior predictive check has been added to the original data frame</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts.ppc</span> </code></pre></div>
<pre><code>## # A tibble: 15 Ã— 4
##    symbol   `sample wise data` `ppc samples failed` `tot deleterious outliers`
##    &lt;chr&gt;    &lt;list&gt;                            &lt;int&gt;                      &lt;int&gt;
##  1 SLC16A12 &lt;tibble [21 Ã— 14]&gt;                    0                          0
##  2 CYP1A1   &lt;tibble [21 Ã— 14]&gt;                    1                          1
##  3 ART3     &lt;tibble [21 Ã— 14]&gt;                    0                          0
##  4 DIO2     &lt;tibble [21 Ã— 14]&gt;                    0                          0
##  5 OR51E2   &lt;tibble [21 Ã— 14]&gt;                    0                          0
##  6 MUC16    &lt;tibble [21 Ã— 14]&gt;                    0                          0
##  7 CCNA1    &lt;tibble [21 Ã— 14]&gt;                    0                          0
##  8 LYZ      &lt;tibble [21 Ã— 14]&gt;                    1                          1
##  9 PPM1H    &lt;tibble [21 Ã— 14]&gt;                    0                          0
## 10 SUSD5    &lt;tibble [21 Ã— 14]&gt;                    0                          0
## 11 TPRG1    &lt;tibble [21 Ã— 14]&gt;                    0                          0
## 12 EPB42    &lt;tibble [21 Ã— 14]&gt;                    0                          0
## 13 LRRC38   &lt;tibble [21 Ã— 14]&gt;                    0                          0
## 14 SUSD4    &lt;tibble [21 Ã— 14]&gt;                    0                          0
## 15 MMP8     &lt;tibble [21 Ã— 14]&gt;                    1                          1</code></pre>
<p>The new data frame contains plots for each gene</p>
<p>We can visualise the top five differentially transcribed genes</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts.ppc_plots</span> <span class="op">=</span> 
    <span class="va">counts.ppc</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="../reference/plot_credible_intervals.html">plot_credible_intervals</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts.ppc_plots</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">plot</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="introduction_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<pre><code>## 
## [[2]]</code></pre>
<p><img src="introduction_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ppcseq_0.99.16   dplyr_1.0.7      knitr_1.33       BiocStyle_2.21.3
## 
## loaded via a namespace (and not attached):
##   [1] matrixStats_0.60.0    fs_1.5.0              httr_1.4.2           
##   [4] doParallel_1.0.16     rprojroot_2.0.2       rstan_2.21.2         
##   [7] tensorA_0.36.2        tools_4.1.0           backports_1.2.1      
##  [10] bslib_0.2.5.1         utf8_1.2.2            R6_2.5.0             
##  [13] colorspace_2.0-2      ggdist_3.0.0          withr_2.4.2          
##  [16] tidyselect_1.1.1      gridExtra_2.3         prettyunits_1.1.1    
##  [19] processx_3.5.2        curl_4.3.2            compiler_4.1.0       
##  [22] textshaping_0.3.5     cli_3.0.1             arrayhelpers_1.1-0   
##  [25] desc_1.3.0            labeling_0.4.2        bookdown_0.22        
##  [28] posterior_1.0.1       sass_0.4.0            scales_1.1.1         
##  [31] checkmate_2.0.0       callr_3.7.0           pkgdown_1.6.1.9001   
##  [34] systemfonts_1.0.2     stringr_1.4.0         digest_0.6.27        
##  [37] StanHeaders_2.21.0-7  rmarkdown_2.10        benchmarkmeData_1.0.4
##  [40] pkgconfig_2.0.3       htmltools_0.5.1.1     parallelly_1.27.0    
##  [43] highr_0.9             fastmap_1.1.0         limma_3.49.4         
##  [46] rlang_0.4.11          jquerylib_0.1.4       farver_2.1.0         
##  [49] generics_0.1.0        svUnit_1.0.6          jsonlite_1.7.2       
##  [52] distributional_0.2.2  inline_0.3.19         magrittr_2.0.1       
##  [55] loo_2.4.1             Matrix_1.3-4          Rcpp_1.0.7           
##  [58] munsell_0.5.0         fansi_0.5.0           abind_1.4-5          
##  [61] furrr_0.2.3           lifecycle_1.0.0       stringi_1.7.3        
##  [64] yaml_2.2.1            edgeR_3.35.0          pkgbuild_1.2.0       
##  [67] grid_4.1.0            listenv_0.8.0         parallel_4.1.0       
##  [70] crayon_1.4.1          lattice_0.20-44       locfit_1.5-9.4       
##  [73] ps_1.6.0              pillar_1.6.2          codetools_0.2-18     
##  [76] stats4_4.1.0          rstantools_2.1.1      glue_1.4.2           
##  [79] evaluate_0.14         V8_3.4.2              BiocManager_1.30.16  
##  [82] RcppParallel_5.1.4    vctrs_0.3.8           foreach_1.5.1        
##  [85] gtable_0.3.0          purrr_0.3.4           tidyr_1.1.3          
##  [88] future_1.21.0         benchmarkme_1.0.7     cachem_1.0.5         
##  [91] ggplot2_3.3.5         xfun_0.25             coda_0.19-4          
##  [94] ragg_1.1.3            tibble_3.1.3          iterators_1.0.13     
##  [97] memoise_2.0.0         globals_0.14.0        tidybayes_3.0.0      
## [100] ellipsis_0.3.2</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Stefano Mangiola.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
